kratos/elements/distance_calculation_flux_based_element.cpp

```diff
@@ -0,0 +1,502 @@
+
+
+    const unsigned int number_of_gauss_points = rGeometry.IntegrationPointsNumber(integration_method);
+    const GeometryType::IntegrationPointsArrayType& integration_points = rGeometry.IntegrationPoints(integration_method);
+
+    for (unsigned int g = 0; g < number_of_gauss_points; ++g){
+        rGaussWeights[g] = det_j_vector[g] * integration_points[g].Weight();
+        rDN_DXContainer[g] = DN_DXContainer_unbounded[g];
+    }
+}
+
+
+//////////////////////////////////////////////////////////////////////////////
+// Public Operations
+
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement<TDim, TNumNodes >::CalculateLocalSystem(
+    MatrixType &rLeftHandSideMatrix,
+    VectorType &rRightHandSideVector,
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Resize and intialize output
+    if (rLeftHandSideMatrix.size1() != LocalSize)
+        rLeftHandSideMatrix.resize(LocalSize, LocalSize, false);
+
+    if (rRightHandSideVector.size() != LocalSize)
+        rRightHandSideVector.resize(LocalSize, false);
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+    noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+
+    const unsigned int step = rCurrentProcessInfo[FRACTIONAL_STEP];
+    
+    if(step == 1){ //solve a transcient diffusion problem
+        CalculatePotentialFlowSystem(rLeftHandSideMatrix, 
+                                     rRightHandSideVector,
+                                     rCurrentProcessInfo);
+    }
+    else if (step == 2){ // solve convection + source
+        CalculateDistanceSystem(rLeftHandSideMatrix, 
+                                     rRightHandSideVector,
+                                     rCurrentProcessInfo);
+    }
+    else{
+        KRATOS_ERROR << "FRACTIONAL_STEP must be 1 or 2" << std::endl;
+    }
+
+    KRATOS_CATCH("");
+}
+
+
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement<TDim, TNumNodes >::CalculatePotentialFlowSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo)
+{
+    BoundedVector<double,TNumNodes> gauss_weights;
+    BoundedMatrix<double,TNumNodes,TNumNodes> N_container;
+    array_1d<BoundedMatrix<double, TNumNodes, TDim>, TNumNodes> DN_DX_container;
+
+    CalculateGaussPointsData( GetGeometry(), gauss_weights, N_container, DN_DX_container);
+
+    array_1d<double, NumNodes > nodal_values;
+    for(unsigned int i=0; i<NumNodes; i++){
+        nodal_values[i] = GetGeometry()[i].FastGetSolutionStepValue(DISTANCE);
+    }
+
+    //computing density so that Fourier number of the complete domain = 1
+    //this ensures that all regions notice the effect of the fixed nodes.
+    const double domain_length = rCurrentProcessInfo[CHARACTERISTIC_LENGTH];
+    const double density  = 1.0/ (domain_length*domain_length);
```

Comment:
A check for non-zero `CHARACTERISTIC_LENGTH` must be added. As far as I understand, this is not expected to change during the simulation so I'd add it in the `Check` function to do it once. Note that this requires setting this value at the time that the solver `Initialize` happens.

Reply:
ok, added in the Check